lib/iris/tests/unit/experimental/ugrid/test_load_meshes.py

```diff
@@ -0,0 +1,218 @@
+        self.assertEqual(1, len(files))
+        file_meshes = meshes[files[0]]
+        self.assertEqual(1, len(file_meshes))
+        mesh = file_meshes[0]
+        self.assertEqual("mesh", mesh.var_name)
+
+    def test_var_name(self):
+        nc_path = cdl_to_nc(self.ref_cdl)
+        with PARSE_UGRID_ON_LOAD.context():
+            meshes = load_meshes(nc_path, "some_other_mesh")
+        self.assertDictEqual({}, meshes)
+
+    def test_multi_files(self):
+        files_count = 3
+        nc_paths = [cdl_to_nc(self.ref_cdl) for _ in range(files_count)]
+        with PARSE_UGRID_ON_LOAD.context():
+            meshes = load_meshes(nc_paths)
+        self.assertEqual(files_count, len(meshes))
+
+    def test_multi_meshes(self):
+        cdl_extra = """
+                    int mesh2 ;
+                        mesh2:cf_role = "mesh_topology" ;
+                        mesh2:topology_dimension = 2 ;
+                        mesh2:node_coordinates = "node_x node_y" ;
+                        mesh2:face_coordinates = "face_x face_y" ;
+                        mesh2:face_node_connectivity = "face_nodes" ;
+            """
+        vars_string = "variables:"
+        vars_start = self.ref_cdl.index(vars_string) + len(vars_string)
+        ref_cdl = (
+            self.ref_cdl[:vars_start] + cdl_extra + self.ref_cdl[vars_start:]
+        )
+
+        nc_path = cdl_to_nc(ref_cdl)
+        with PARSE_UGRID_ON_LOAD.context():
+            meshes = load_meshes(nc_path)
+
+        files = list(meshes.keys())
+        self.assertEqual(1, len(files))
+        file_meshes = meshes[files[0]]
+        self.assertEqual(2, len(file_meshes))
+        mesh_names = [mesh.var_name for mesh in file_meshes]
+        self.assertIn("mesh", mesh_names)
+        self.assertIn("mesh2", mesh_names)
+
+    def test_no_parsing(self):
+        nc_path = cdl_to_nc(self.ref_cdl)
+        with self.assertRaisesRegex(
+            ValueError, ".*Must be True to enable mesh loading."
+        ):
+            _ = load_meshes(nc_path)
+
+    def test_invalid_scheme(self):
+        with self.assertRaisesRegex(
+            ValueError, "Iris cannot handle the URI scheme:.*"
+        ):
+            with PARSE_UGRID_ON_LOAD.context():
+                _ = load_meshes("foo://bar")
+
+    @tests.skip_data
+    def test_non_nc(self):
+        log_regex = r"Ignoring non-NetCDF file:.*"
+        with self.assertLogs(logger, level="INFO", msg_regex=log_regex):
+            with PARSE_UGRID_ON_LOAD.context():
+                meshes = load_meshes(
+                    tests.get_data_path(["PP", "simple_pp", "global.pp"])
+                )
+        self.assertDictEqual({}, meshes)
+
+
+class TestsHttp(tests.IrisTest):
+    # Tests of HTTP (OpenDAP) loading need mocking since we can't have tests
+    #  that rely on 3rd party servers.
+    def setUp(self):
+        patcher = mock.patch("iris.fileformats.FORMAT_AGENT.get_spec")
+        self.addCleanup(patcher.stop)
```

Comment:
See previous note on using `self.patch` for this.

Reply:
436f5f737bc1c1b8fc08631717bcf64fab5f054e